tchannel/tornado/peer.py

```diff
@@ -624,7 +624,7 @@ def send_with_retry(self, request, peer, attempt_times, retry_delay):
                 )
 
                 if not connection:
-                    raise
+                    raise protocol_error
```

Comment:
To explain that further:

    try:
        foo()
    except Exception:
        # stuff
        raise

Usually works, but if you're using coroutines and yield one inside the Exception block, and the called coroutine returns with `raise Return`, the system's knowledge of "the last exception" changes.